crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -59,30 +67,52 @@ enum AuctionIdRecoveryStatus {
 }
 
 impl OnSettlementEventUpdater {
-    pub async fn run_forever(self) -> ! {
-        let mut current_block = self.eth.current_block().borrow().to_owned();
-        let mut block_stream = ethrpc::current_block::into_stream(self.eth.current_block().clone());
+    pub fn new(eth: infra::Ethereum, db: Postgres) -> Self {
+        let inner = Arc::new(Inner {
+            eth,
+            db,
+            notify: Notify::new(),
+        });
+        let inner_clone = inner.clone();
+        tokio::spawn(async move { Inner::wait_for_updates(inner_clone).await });
+        Self { inner }
+    }
+
+    /// Deletes settlement_observations and order executions for the given range
+    pub async fn delete_observations(
+        transaction: &mut PgTransaction<'_>,
+        range: RangeInclusive<u64>,
+    ) -> Result<()> {
+        database::settlements::delete(transaction, *range.start() as i64)
+            .await
+            .context("delete_settlement_observations")?;
+
+        Ok(())
+    }
+
+    /// Schedules an update loop on a background thread
+    pub fn schedule_update(&self) {
+        self.inner.notify.notify_one();
+    }
+}
+
+impl Inner {
+    async fn wait_for_updates(self: Arc<Inner>) -> ! {
```

Comment:
`wait_for_updates` seems like a poor name. That name would suggest to me that this function returns when some update happened.